src/expression.js

```diff
@@ -206,7 +206,7 @@ pp.parseExprSubscripts = function(refShorthandDefaultPos) {
   let startPos = this.start, startLoc = this.startLoc
   let expr = this.parseExprAtom(refShorthandDefaultPos)
   if (refShorthandDefaultPos && refShorthandDefaultPos.start) return expr
-  return this.parseSubscripts(expr, startPos, startLoc)
+  return this.parseSubscripts(expr, startPos, startLoc, expr.type === "ArrowFunctionExpression" && this.parenthesesCount === 0)
```

Comment:
This is handling any case for call subscript (`(`) right after unparenthesized arrow function. But for `foo => bar()` if you put another `(` right after it, it will be parsed as body of arrow function.

E.g. `foo => bar()()` will be correctly parsed with `bar()()` as the body, right? In that case our problem is irrelevant and check will pass.

So yes, it becomes relevant only when body is a block.